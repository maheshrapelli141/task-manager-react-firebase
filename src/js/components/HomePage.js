import React from 'react';
import TaskManagement from '../../TaskManagement.jpg';

const HomePage = () => (
  <img src={TaskManagement} alt="Why task management?" width="100%" />
);

export default HomePage;
