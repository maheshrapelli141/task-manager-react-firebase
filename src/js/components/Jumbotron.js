import React from 'react';
import PropTypes from 'prop-types';

const Jumbotron = ({ children }) => (
  <div className="jumbotron jumbotron-fluid">
    <div className="container">{children}</div>
  </div>
);

Jumbotron.propTypes = {
  children: PropTypes.node.isRequired
};

export default Jumbotron;
