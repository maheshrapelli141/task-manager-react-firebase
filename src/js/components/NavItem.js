import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const NavItem = ({ path, label }) => (
  <li className="nav-item">
    <NavLink to={path} activeClassName="active" className="nav-link">
      {label}
    </NavLink>
  </li>
);

NavItem.propTypes = {
  path: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default NavItem;
