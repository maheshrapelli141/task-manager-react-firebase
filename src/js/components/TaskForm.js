import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { addTask } from '../actions/tasks.actions';
import { toDateString } from '../util/DateUtil';
import tasksService from '../services/task.service';

const initialState = {
  name: '',
  category: '',
  description: '',
  dueDate: '',
  reminderDate: ''
};

export class TaskForm extends Component {
  constructor(props) {
    super(props);

    this.state = initialState;

    this.onInputChanged = this.onInputChanged.bind(this);
    this.onFormSubmit = this.onFormSubmit.bind(this);
  }

  onInputChanged(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  onFormSubmit(event) {
    event.preventDefault();

    tasksService.addTask(this.state).then(res => {
      console.log({res});
      this.props.addTask(this.state);
      this.setState(initialState);
    }).catch(err => {
      console.log({err});
    })
  }

  isValidForm() {
    const { name, category, dueDate } = this.state;
    return name.length && category.length && dueDate.length;
  }

  render() {
    const { name, category, description, reminderDate, dueDate } = this.state;
    const minDate = toDateString(Date.now());
    return (
      <section>
        <div className="card">
          <div className="card-body">
            <form onSubmit={this.onFormSubmit}>
              <div className="form-group required">
                <label htmlFor="name">Name</label>
                <input
                  type="text"
                  className="form-control"
                  id="name"
                  value={name}
                  onChange={this.onInputChanged}
                  required
                />
              </div>
              <div className="form-group required">
                <label htmlFor="name">Category</label>
                <input
                  type="text"
                  className="form-control"
                  id="category"
                  value={category}
                  onChange={this.onInputChanged}
                  required
                />
              </div>
              <div className="form-group required">
                <label htmlFor="dueDate">Due date</label>
                <input
                  type="date"
                  className="form-control"
                  id="dueDate"
                  min={minDate}
                  value={dueDate}
                  onChange={this.onInputChanged}
                  required
                />
              </div>
              <div className="form-group">
                <label htmlFor="reminderDate">Reminder date</label>
                <input
                  type="date"
                  className="form-control"
                  id="reminderDate"
                  min={minDate}
                  value={reminderDate}
                  onChange={this.onInputChanged}
                />
              </div>
              <div className="form-group">
                <label htmlFor="description" className="bmd-label-floating">
                  Description
                </label>
                <textarea
                  className="form-control"
                  id="description"
                  rows="3"
                  value={description}
                  onChange={this.onInputChanged}
                />
              </div>
              <button
                type="submit"
                className="btn btn-primary btn-lg btn-block btn-raised"
                disabled={!this.isValidForm()}
              >
                Add Task
              </button>
            </form>
          </div>
        </div>
      </section>
    );
  }
}

TaskForm.propTypes = {
  addTask: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => {
  return {
    addTask: task => dispatch(addTask(task))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(TaskForm);
