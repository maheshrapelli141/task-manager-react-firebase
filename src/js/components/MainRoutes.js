import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import HomePage from './HomePage';
import TaskPage from './TaskPage';
import ReportPage from './ReportPage';
import PrivateRoute from './PrivateRoute';

const MainRoutes = () => (
  <Switch>
    <Route exact path="/" component={HomePage} />
    <PrivateRoute path="/task" component={TaskPage} />
    <PrivateRoute path="/report" component={ReportPage} />
    <Redirect to="/" />
  </Switch>
);

export default MainRoutes;
