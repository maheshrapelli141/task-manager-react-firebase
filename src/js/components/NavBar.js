import React from 'react';
import { connect } from 'react-redux';

import NavItem from './NavItem';
import { auth, provider } from '../util/firebase';

const login = () => { auth.signInWithPopup(provider) }
const logout = () => { auth.signOut() }

const NavBar = props => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
    <a className="navbar-brand" href="#">
      Navbar
    </a>
    <button
      className="navbar-toggler"
      type="button"
      data-toggle="collapse"
      data-target="#navbarContent"
    >
      <span className="navbar-toggler-icon" />
    </button>

    <div className="collapse navbar-collapse" id="navbarContent">
      {props.user
        ?
          <ul className="navbar-nav mr-auto">
            <NavItem path="/task" label="Task" />
            <NavItem path="/report" label="Report" />
            <button type="button" className="btn btn-primary btn-right" onClick={logout}  >Logout</button>
          </ul>
        :
          <ul className="navbar-nav mr-auto">
            <button type="button" className="btn btn-primary btn-right" onClick={login}  >Login</button>
          </ul>
      }

    </div>
  </nav>
);

const mapStateToProps = state => {
  const { user } = state;
  return {
    user
  };
}

export default connect(mapStateToProps)(NavBar);
