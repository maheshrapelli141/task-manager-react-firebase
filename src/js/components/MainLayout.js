import React from 'react';
import { HashRouter } from 'react-router-dom';

import NavBar from './NavBar';
import MainRoutes from './MainRoutes';

const MainLayout = () => (
  <HashRouter hashType="noslash">
    <div className="wrapper">
      <NavBar />
      <main>
        <div className="card">
          <div className="card-body">
            <MainRoutes />
          </div>
        </div>
      </main>
    </div>
  </HashRouter>
);

export default MainLayout;
