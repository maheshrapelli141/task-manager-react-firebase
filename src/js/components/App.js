import React from 'react';
import { connect } from 'react-redux';
import { auth } from '../util/firebase';
import MainLayout from './MainLayout';
import { login } from '../actions/user.actions';

class App extends React.Component {
  constructor(props) {
    super(props);
    
  }

  componentDidMount() {
    auth.onAuthStateChanged((user) => {
      this.props.dispatch(login(user)) 
    });
  }


  render() {
    return (
      <div id="mainContainer">
        <MainLayout />
      </div>);
  }
}
const mapStateToProps = state => {
  const { user } = state;
  return {
    user
  }
};
export default connect(mapStateToProps)(App);
