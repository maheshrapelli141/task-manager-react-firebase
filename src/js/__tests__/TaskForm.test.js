import React from 'react';
import { mount } from 'enzyme';

import { TaskForm } from '../components/TaskForm';

Date.now = jest.fn().mockReturnValue('2018-01-01');

const setup = () => {
  const props = {
    addTask: jest.fn()
  };

  const wrapper = mount(<TaskForm {...props} />);

  return {
    props,
    wrapper
  };
};

describe('TaskForm component', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should disable add task button when mandatory fields are empty', () => {
    const { wrapper } = setup();
    expect(wrapper.find('button').props().disabled).toBeTruthy();
  });

  it('shoud call addTask when form is submitted', () => {
    const { wrapper, props } = setup();
    expect(props.addTask.mock.calls).toHaveLength(0);
    wrapper.find('button').simulate('submit');
    expect(props.addTask.mock.calls).toHaveLength(1);
    wrapper.find('form').simulate('submit');
    expect(props.addTask.mock.calls).toHaveLength(2);
  });
});
