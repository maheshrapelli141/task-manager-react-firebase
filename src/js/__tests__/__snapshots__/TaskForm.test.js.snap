// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm component should render correctly 1`] = `
<TaskForm
  addTask={[MockFunction]}
>
  <section>
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group required"
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="form-control"
              id="name"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group required"
          >
            <label
              htmlFor="name"
            >
              Category
            </label>
            <input
              className="form-control"
              id="category"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group required"
          >
            <label
              htmlFor="dueDate"
            >
              Due date
            </label>
            <input
              className="form-control"
              id="dueDate"
              min="2018-01-01"
              onChange={[Function]}
              required={true}
              type="date"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="reminderDate"
            >
              Reminder date
            </label>
            <input
              className="form-control"
              id="reminderDate"
              min="2018-01-01"
              onChange={[Function]}
              type="date"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="bmd-label-floating"
              htmlFor="description"
            >
              Description
            </label>
            <textarea
              className="form-control"
              id="description"
              onChange={[Function]}
              rows="3"
              value=""
            />
          </div>
          <button
            className="btn btn-primary btn-lg btn-block btn-raised"
            disabled={true}
            type="submit"
          >
            Add Task
          </button>
        </form>
      </div>
    </div>
  </section>
</TaskForm>
`;
